server/src/lib/model/db.ts

```diff
@@ -170,22 +218,103 @@ export default class DB {
             FROM clips
             WHERE clips.original_sentence_id = sentences.id AND
                   clips.client_id = ?
+          ) AND NOT EXISTS (
+            SELECT *
+            FROM skipped_sentences skipped
+            WHERE skipped.sentence_id = sentences.id AND
+              skipped.client_id = ?
```

Comment:
As discussed, added skipped_sentence handling to the regular sentence serving logic too.